tests/test_hf_api.py

```diff
@@ -864,15 +850,17 @@ def test_list_models_author(self):
         models = _api.list_models(author="google")
         self.assertGreater(len(models), 10)
         self.assertIsInstance(models[0], ModelInfo)
-        [self.assertTrue("google" in model.author for model in models)]
+        for model in models:
+            self.assertTrue("google" in model.author)
```

Comment:
@SBrandeis @LysandreJik @julien-c It seems that `flake8-bugbear` actually helped to find a bug here.

This line (introduced in https://github.com/huggingface/huggingface_hub/pull/531)
```py
[self.assertTrue("google" in model.author for model in models)]
```
was most likely intended to be written
```py
[self.assertTrue("google" in model.author) for model in models]
```
In the end, the `self.assertTrue` is not evaluating the value of each author but the generator `("google" in model.author for model in models)` which is always True.

----
Jumping to our actual problem now,  the field `model.author` is currently set to `None` for all the models returned by the api. Here is an example of `ModelInfo` returned by `_api.list_models(author="google")` (out of 515 of them, all without authors).

```py
# https://huggingface.co/google/t5-11b-ssm-wqo
 ModelInfo: {
        modelId: google/t5-11b-ssm-wqo
        sha: None
        lastModified: None
        tags: None
        pipeline_tag: None
        siblings: None
        private: False
        author: None
        config: None
        id: google/t5-11b-ssm-wqo
}
```

I'll investigate more why the authors are not set to "google". If you have any quick insight, I am interested.

Reply:
[TO BE CONFIRMED](https://huggingface.slack.com/archives/C023JAKTR2P/p1659709357156959) 

I "fixed" the test by testing against model id:
```py
for model in models:
    self.assertTrue(model.modelId.startswith("google/"))
```

Reply:
i think the issue might be that `/api/models` does not output the full model objects and might not have a `author` attribute by default

For this, I would suggest reading the server code (file named `apiListingRoutes.ts`), we can help you read it, cc @SBrandeis

Reply:
@julien-c yes I looked at the response body and `authpr` is indeed not returned on the `/api/models` endpoint. But here I'd say it's really not a problem, we just wanted a way to test that only models from the `google` author are returned, which is the case.

Reply:
yes 👍